File: src/test/java/kitchenpos/ordertable/acceptance/OrderTableAcceptanceTest.java
package kitchenpos.ordertable.acceptance;

import kitchenpos.support.AcceptanceTest;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class OrderTableAcceptanceTest extends AcceptanceTest {

    /**
     * <pre>
     * when 테이블을 등록하면
     * then 테이블 목록 조회 시 등록한 테이블을 찾을 수 있다.
     * then 등록된 테이블은 현재 비어있는 테이블이어야 한다.
     * </pre>
     */
    @Test
    @DisplayName("등록")
    void create() {

    }

    /**
     * <pre>
     * given 2개의 테이블을 등록하고
     * when  테이블 목록을 조회하면
     * then  등록한 2개의 테이블을 찾을 수 있다.
     * </pre>
     */
    @Test
    @DisplayName("목록조회")
    void findAll() {

    }

    /**
     * <pre>
     * given 테이블을 등록하고
     * when  손님이 테이블에 앉으면
     * then  해당 테이블은 점유 상태이다.
     * </pre>
     */
    @Test
    @DisplayName("점유")
    void sit() {

    }

    /**
     * <pre>
     * given 테이블을 등록하고
     * given 해당 테이블에 손님이 앉았다가
     * when  식사를 다 하고 나가서 정리하면
     * then  해당 테이블은 비어있으며 앉을 수 있다.
     * </pre>
     */
    @Test
    @DisplayName("정리")
    void clear() {

    }

    /**
     * <pre>
     * given 테이블을 등록하고
     * given 해당 테이블에 손님이 않아
     * when  테이블에 앉은 손님 수를 변경하면
     * then  테이블 목록 조회 시 해당 테이블에 앉아 있는 손님 수는 변경한 손님 수와 동일하다.
     * </pre>
     */
    @Test
    @DisplayName("손님 수 변경")
    void changeNumberOfGuests() {

    }

}

File: src/test/kotlin/kitchenpos/acceptance/ordertable/OrderTableAcceptanceTest.kt
package kitchenpos.acceptance.ordertable

import domain.MenuFixtures.makeMenuOne
import domain.MenuGroupFixtures.makeMenuGroupOne
import domain.OrderFixtures.makeOrderTwo
import domain.OrderTableFixtures.makeOrderTableOne
import domain.ProductFixtures.makeProductOne
import io.kotest.matchers.shouldBe
import kitchenpos.acceptance.CommonAcceptanceTest
import kitchenpos.domain.MenuGroupRepository
import kitchenpos.domain.MenuRepository
import kitchenpos.domain.OrderRepository
import kitchenpos.domain.OrderTableRepository
import kitchenpos.domain.ProductRepository
import org.springframework.boot.test.web.server.LocalServerPort

class OrderTableAcceptanceTest(
    @LocalServerPort
    override val port: Int,
    private val menuRepository: MenuRepository,
    private val menuGroupRepository: MenuGroupRepository,
    private val productRepository: ProductRepository,
    private val orderRepository: OrderRepository,
    private val orderTableRepository: OrderTableRepository,
) : CommonAcceptanceTest() {
    init {
        given("가게 사장님이 로그인한 뒤에") {
            `when`("주문 테이블을 생성하면") {
                then("주문 테이블이 생성된다.") {
                    val response =
                        commonRequestSpec()
                            .given()
                            .body(
                                """
                                {
                                      "name": "9번"
                                }
                                """.trimIndent(),
                            )
                            .post("/api/order-tables")
                            .then()
                            .log().everything()
                            .extract().response()

                    val jsonPathEvaluator = response.jsonPath()

                    jsonPathEvaluator.getInt("numberOfGuests") shouldBe 0
                    response.statusCode shouldBe 201
                }
            }

            `when`("주문 테이블에 손님이 앉으면") {
                then("주문 테이블에 손님이 앉는다.") {
                    val orderTable = orderTableRepository.save(makeOrderTableOne())

                    val response =
                        commonRequestSpec()
                            .given()
                            .put("/api/order-tables/${orderTable.id}/sit")
                            .then()
                            .log().everything()
                            .extract().response()

                    val jsonPathEvaluator = response.jsonPath()

                    jsonPathEvaluator.getBoolean("occupied") shouldBe true
                    response.statusCode shouldBe 200
                }
            }

            `when`("주문 테이블을 비우면") {
                then("주문 테이블이 비워진다.") {
                    orderTableRepository.save(makeOrderTableOne())
                    productRepository.save(makeProductOne())
                    menuGroupRepository.save(makeMenuGroupOne())
                    menuRepository.save(makeMenuOne())
                    orderRepository.save(makeOrderTwo())

                    val response =
                        commonRequestSpec()
                            .given()
                            .put("/api/order-tables/6924640b-b0fc-4c86-84f9-b750eeba0205/clear")
                            .then()
                            .log().everything()
                            .extract().response()

                    val jsonPathEvaluator = response.jsonPath()

                    jsonPathEvaluator.getBoolean("occupied") shouldBe false
                    response.statusCode shouldBe 200
                }
            }
//
//            `when`("주문 테이블의 손님 수를 변경하면") {
//                then("주문 테이블의 손님 수가 변경된다.") {
//                    val orderTableId = createOrderTable()
//
//                    commonRequestSpec()
//                        .given()
//                        .put("/api/order-tables/$orderTableId/sit")
//                        .then()
//                        .log().everything()
//                        .extract().response()
//                }
//            }
        }
    }
}
